mediator/med_diag_mod.F90

```diff
@@ -1306,18 +1456,26 @@ subroutine diag_ocn(FB, fldname, nf, ic, areas, frac, budget, rc)
       real(r8)               , intent(in)    :: areas(:)
       real(r8)               , intent(in)    :: frac(:)
       real(r8)               , intent(inout) :: budget(:,:,:)
+      real(r8), optional     , intent(in)    :: scale
       integer                , intent(out)   :: rc
       ! local variables
       integer           :: n, ip
+      type(ESMF_field)  :: lfield
       real(r8), pointer :: data(:) => null()
       ! ------------------------------------------------------------------
       rc = ESMF_SUCCESS
       if ( FB_fldchk(FB, trim(fldname), rc=rc)) then
-         call FB_GetFldPtr(FB, trim(fldname), fldptr1=data, rc=rc)
+         call ESMF_FieldBundleGet(FB, trim(fldname), field=lfield, rc=rc)
+         if (ChkErr(rc,__LINE__,u_FILE_u)) return
+         call ESMF_FieldGet(lfield, farrayptr=data, rc=rc)
          if (ChkErr(rc,__LINE__,u_FILE_u)) return
          ip = period_inst
          do n = 1, size(data)
-            budget(nf,ic,ip) = budget(nf,ic,ip) + areas(n)*frac(n)*data(n)
+            if (present(scale)) then
```

Comment:
It may be faster here to define a new variable lscale and do
```
if(present(scale)) then
  lscale = scale
else
  lscale = 1.0R8
endif
```
in this way you avoid the if inside the loop which prevents vectorization.